add option for using subset of training chips

simonkassel opened the pull request:
This PR adds the option to train a model using only a subset of the available training chips. This should be a value between 0 and 1 and specifies what proportion of the training data to use. It defaults to 1. 

- I used the name `train_proportion`. Don't love it, it's a little long, would be interested in hearing a different idea
- I added an example that uses this parameter to the potsdam examples script. I can take it out but I just left it in there for now for easy testing.
- Should this also have an option to select a number of total chips to use rather than a proportion of the total?

### To Test:

I have tested it locally and remotely on the potsdam dataset

to run a test, adapt the basic fastai semseg example:

```
export RAW_URI="s3://raster-vision-raw-data/isprs-potsdam"
export PROCESSED_URI="s3://raster-vision-lf-dev/fastai/potsdam/processed-data"
export ROOT_URI="{insert root uri}"

rastervision -p fastai run aws_batch -e examples.semantic_segmentation.potsdam -m \
*exp_half_train_data* -a raw_uri $RAW_URI -a processed_uri $PROCESSED_URI\ 
-a root_uri $ROOT_URI -a test False --splits 4
```

closes #12

lewfish commented on fastai_plugin/semantic_segmentation_backend.py:
```diff
@@ -154,6 +155,42 @@ def _write_zip(results, split):
+
+        all_train_uri = join(chip_dir, 'train-img')
+        all_train = os.listdir(all_train_uri)
+        sample_size = round(prop * len(all_train))
```
I would also sort `all_train` before doing the sampling to make things repeatable. This is because `os.listdir` returns things in an arbitrary order.

simonkassel commented:
These all sound good. I made the changes. The most significant change is that I added an optional `train_count` parameter. I prioritized `train_count` over `train_prop` so if `train_count` is set it uses that otherwise it defaults to `train_prop`. I added an additional potsdam example that uses the `train_count` parameter.

Here is the updated command to run both the `train_count` and `train_prop` potsdam experiments: `rastervision -p fastai run aws_batch -e examples.semantic_segmentation.potsdam -m *exp_subset* -a raw_uri $RAW_URI -a processed_uri $PROCESSED_URI -a root_uri $ROOT_URI --splits 4`
